django_redis/client/default.py

```diff
@@ -185,6 +185,8 @@ def add(self, key, value, timeout=DEFAULT_TIMEOUT, version=None, client=None):
 
         Returns ``True`` if the object was added, ``False`` if not.
         """
+        if self.has_key(key):
+            return False
         return self.set(key, value, timeout, version=version, client=client, nx=True)
```

Comment:
There is likely a race condition between the `has_key` & `set` call. Maybe this should use the Redis `SETNX` command instead.

https://redis.io/commands/setnx

Reply:
Agree, the current proposed approach can cause race conditions.

Reply:
yeah, seems like the nx should be default option

Reply:
Why do we need the `.has_key()` at all then? Shouldn't `.set(..., nx=True)` be enough? Why add additional round trips to Redis?

Reply:
`.set(..., nx=True)` isn't returning False like it should. Unfortunately I don't have enough time to look more into this until the weekend. It looked like a quick fix initially, but evidently it's not.

My break time is up, but I looked briefly into it: `redis-py`'s `setnx` returns False when it fails, and when the `nx` flag is set it calls

https://github.com/niwinz/django-redis/blob/f411c553a7749013e4879ebf08c93daf222346f2/django_redis/client/default.py#L137

So it's probably something here that's not behaving as it should. The code of `has_key` looks fine to me though.